src/execution/execute-afm.ts

```diff
@@ -1,137 +1,251 @@
+        return this.xhr.post(`/gdc/app/projects/${projectId}/executeAfm`, { body: JSON.stringify(execution) })
+            .then(apiResponse => apiResponse.getData())
+            .then(unwrapExecutionResponse);
     }
 
-    private getOnePage(
-        pollingUri: string, offset: number[], prevResultWrapper?: Execution.IExecutionResultWrapper
-    ): any {
-        return this.fetchExecutionResult(pollingUri, offset)
+    /**
+     * Get whole (all pages) Result from Execution
+     *
+     * @method getWholeExecutionResult
+     * @param {string} executionResultUri
+     *
+     * @returns {Promise<Execution.IExecutionResult | null>}
+     *  Promise with `executionResult` or `null` (null means empty response - HTTP 204)
+     *  See https://github.com/gooddata/gooddata-typings/blob/master/src/Execution.ts#L88
+     */
+    private getWholeExecutionResult(executionResultUri: string)
+        : Promise<Execution.IExecutionResult | null> {
+        const numOfDimensions = Number(qs.parse(executionResultUri).dimensions);
+        validateNumOfDimensions(numOfDimensions);
+
+        const limit = Array(numOfDimensions).fill(DEFAULT_LIMIT);
+        const offset = Array(numOfDimensions).fill(0);
+
+        return this.getPage(true, executionResultUri, limit, offset);
+    }
+
+    private getPage(
+        recursive: boolean,
+        executionResultUri: string,
+        limit: number[],
+        offset: number[],
+        prevExecutionResult?: Execution.IExecutionResult
+    ): Promise<Execution.IExecutionResult | null> {
+        return this.fetchExecutionResult(executionResultUri, limit, offset)
             .then((executionResultWrapper: Execution.IExecutionResultWrapper) => {
                 if (!executionResultWrapper) {
                     return null;
                 }
 
-                const newResultWrapper = prevResultWrapper
-                    ? mergePageData(prevResultWrapper, executionResultWrapper)
-                    : executionResultWrapper;
+                const executionResult = unwrapExecutionResult(executionResultWrapper);
+
+                if (!recursive) {
+                    return executionResult;
+                }
+
+                const newExecutionResult = prevExecutionResult
+                    ? mergePage(prevExecutionResult, executionResult)
+                    : executionResult;
+
+                const { offset, total } = executionResult.paging;
+                const nextOffset = getNextOffset(limit, offset, total);
+                const nextLimit = getNextLimit(limit, nextOffset, total);
+
+                return nextPageExists(nextOffset, total)
+                    ? this.getPage(true, executionResultUri, nextLimit, nextOffset, newExecutionResult)
+                    : newExecutionResult;
+            });
+    }
+
+    private fetchExecutionResult(executionResultUri: string, limit: number[], offset: number[]) {
```

Comment:
Missing return value?

Reply:
`fetchExecutionResult` should be by part of `gooddata-js/src/execution.ts` file

Reply:
fixed